src/test/java/eu/xenit/gradle/docker/compose/DockerComposeConventionImplTest.java

```diff
@@ -228,5 +228,33 @@ public void testFromProjectWithDockerAlfrescoPlugin() {
         assertEquals(DOCKER_IMAGE_ID, composeSettings.getEnvironment().get("SUB_PROJECT1_DOCKER_IMAGE"));
     }
 
+    @Test
+    public void testFromProjectWithoutEnvironmentVariable(){
+        composeConvention.fromProject(null,project);
+        TaskCollection<DockerBuildImage> buildImageTaskCollection = project.getTasks().withType(DockerBuildImage.class);
+
+        assertTaskDependsOn(composeSettings.getUpTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getBuildTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getPushTask(), dockerBuildImageProvider);
+
+        runSetupImageEnvironmentWithComposeUp();
+
+        assertEquals(DOCKER_IMAGE_ID, composeSettings.getEnvironment().get("BUILD_DOCKER_IMAGE_DOCKER_IMAGE"));
+    }
+
+    @Test
+    public void testFromProjectWithEnvironmentVariable(){
+        composeConvention.fromProject("testname",project);
+        TaskCollection<DockerBuildImage> buildImageTaskCollection = project.getTasks().withType(DockerBuildImage.class);
+
+        assertTaskDependsOn(composeSettings.getUpTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getBuildTask(), dockerBuildImageProvider);
+        assertTaskDependsOn(composeSettings.getPushTask(), dockerBuildImageProvider);
+
+        runSetupImageEnvironmentWithComposeUp();
+
+        assertEquals(DOCKER_IMAGE_ID, composeSettings.getEnvironment().get("testname"));
```

Comment:
Okee. Ik wil nog een test toevoegen waarop ik kan checken op een task met een andere naam dan "buildDockerImage"?

Hoe doe ik dat precies? Ik heb gezien dat ik een taak kan registeren en snap hoe dat moet maar ik snap niet hoe ik dit dan kan testen. Mss 10 minuten call morgen?

Reply:
Creating new tasks with name DockerBuildImage can be done with `project.getTasks().register("myBuildTaskName", DockerBuildImage.class)`.

You will need to set up a bogus image ID in the same way as done in `createBuildDockerImage()` in this test class.